src/editor/EditorCommandHandlers.js

```diff
@@ -467,12 +485,29 @@ define(function (require, exports, module) {
             // Comment out - add the suffix to the start and the prefix to the end.
             if (canComment) {
                 var completeLineSel = sel.start.ch === 0 && sel.end.ch === 0 && sel.start.line < sel.end.line;
+                var startCh = _firstNotWs(doc, sel.start.line);
                 if (completeLineSel) {
-                    editGroup.push({text: suffix + "\n", start: sel.end});
-                    editGroup.push({text: prefix + "\n", start: sel.start});
+                    if (indentBlockComment) {
+                        var endCh = _firstNotWs(doc, sel.end.line - 1);
+                        editGroup.push({
+                            text: _.repeat(" ", endCh) + suffix + "\n",
```

Comment:
Can't always insert spaces -- need to use indent preference here in case it's Tabs.